### PR TITLE
Allow generating Dwarf version 5

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Contributes to https://github.com/dotnet/runtime/issues/96848.

Depends on https://github.com/dotnet/llvm-project/pull/161 but doesn't break until the new option is used.

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
I had to make it an option instead of hardcoding it because e.g. GDB that ships with Ubuntu 18 can't read this format.

**Comment by jkotas** [comment]
Are we also going to need to build the runtime itself with dwarf5?

**Review by jkotas — APPROVED** [review]

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/CompilationBuilder.Aot.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ partial class CompilationBuilder
         protected bool _methodBodyFolding;
         protected InstructionSetSupport _instructionSetSupport;
         protected SecurityMitigationOptions _mitigationOptions;
+        protected bool _useDrawf5;
```
Comment (Clockwork-Muse) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/CompilationBuilder.Aot.cs:27]: ...
`protected bool _useDwarf5;`?

**Comment by MichalStrehovsky** [comment]
> Are we also going to need to build the runtime itself with dwarf5?

Possibly. I'll retest end to end once this is merged. I only tested this with ZeroSharp so that I can easily crossbuild out of Windows without having to do any building on Linux (ObjWriter, product, put them together, etc; things were easier when ObjWriter lived in the same repo).

**Pull request merged**